curl/build.sh

```diff
@@ -24,10 +24,12 @@ fi
 #
 # Configure
 #
+# shellcheck disable=SC2046 # Word splitting is expected when adding mbetls-related options
 if [ ! -f "$PREFIX_PORT_BUILD/config.status" ]; then
 	( cd "$PREFIX_PORT_BUILD" && PKG_CONFIG="" "$PREFIX_CURL_SRC/configure" CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS" \
 		--host="${HOST}" --sbindir="$PREFIX_PROG" --disable-pthreads --disable-threaded-resolver \
-		--disable-ipv6 --prefix="$PREFIX_CURL_INSTALL" --disable-ntlm-wb --without-zlib )
+		--disable-ipv6 --prefix="$PREFIX_CURL_INSTALL" --disable-ntlm-wb --without-zlib \
+		$([[ "$PORTS_CURL_USE_MBEDTLS" = "y" ]] && echo " --without-ssl --with-mbedtls "))
```

Comment:
Consider this approach. It also doesn't trigger shellcheck and I think is much more cleaner.
```suggestion
	configure_params=(--host="${HOST}" --sbindir="$PREFIX_PROG" --disable-pthreads --disable-threaded-resolver \
		--disable-ipv6 --prefix="$PREFIX_CURL_INSTALL" --disable-ntlm-wb --without-zlib)
	[[ "$PORTS_CURL_USE_MBEDTLS" = "y" ]] && configure_params+=(--without-ssl --with-mbedtls)
	( cd "$PREFIX_PORT_BUILD" && PKG_CONFIG="" "$PREFIX_CURL_SRC/configure" CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS" "${configure_params[@]}" )
```

Not sure how about compatibility, but I think that simple arrays is in bash from eternity.